src/stirling/source_connectors/socket_tracer/openssl_trace_bpf_test.cc

```diff
@@ -31,12 +31,20 @@
 #include "src/stirling/source_connectors/socket_tracer/testing/protocol_checkers.h"
 #include "src/stirling/source_connectors/socket_tracer/testing/socket_trace_bpf_test_fixture.h"
 #include "src/stirling/testing/common.h"
+#include "src/stirling/source_connectors/socket_tracer/uprobe_symaddrs.h"
 
 namespace px {
 namespace stirling {
 
 namespace http = protocols::http;
 
+bool Init() {
+  FLAGS_openssl_force_raw_fptrs = true;
```

Comment:
Ideally the test suite would be able to run both cases for all of the tests. Any suggestions on how that could be accomplished?

Reply:
@ddelnano, a couple of pointers:
1) You can use `INSTANTIATE_TEST_SUITE_P` which is part of gtest to pass in parameters to a test. You can find plenty of examples in Stirling if you grep around.
2) You can use PL_SET_FOR_SCOPE for assigning flags in tests. Since flags are global, it helps to avoid unintended consequences when changing a flag (i.e. so the flag only applies for the test case in questions).

Reply:
1. I did notice the `INSTANTIATE_TEST_SUITE_P` examples, but the openssl_trace_bpf_test code uses `TYPED_TEST`s. I wasn't sure how to incorporate the parameter aspects when typed tests are in use
2. Thanks for that tip. `PL_SET_FOR_SCOPE` is exactly what I was looking for.

Reply:
This [Stackoverflow post](type-parameterized tests and value-parameterized tests) post mentions that the two features are not complementary and links to a google group [discussion](https://groups.google.com/g/googletestframework/c/3d8H5J5Isnw) with the maintainers. And given the copy/move issues with the test cases due to the underlying ContainerRunner instances, I think we make the typed tests work.

I've crated a simpler example (copied below). I need to figure out how to declare each `TYPED_TEST` multiple times still, which is what I'm trying to accomplish with the `OPENSSL_TYPED_TEST` macro. Please let me know your thoughts on that approach or any other ideas you have.

```
#include <gtest/gtest.h>

class ContainerRunner {
  public:
    void Run() {
    }
};

class ContainerOne : public ContainerRunner { };
class ContainerTwo : public ContainerRunner { };

template <typename TServerContainer, bool TForceFptr>
class BaseOpenSSLTraceTest : public ::testing::Test {
  public:
    BaseOpenSSLTraceTest() {
      server_.Run();
    }
    TServerContainer server_;
    bool force_fptr_ = TForceFptr;

};

template <typename T>
using OpenSSLTraceTest = BaseOpenSSLTraceTest<T, false>;

typedef ::testing::Types<ContainerOne, ContainerTwo>
  OpenSSLServerImplementations;

// Old way that works
TYPED_TEST_SUITE(OpenSSLTraceTest, OpenSSLServerImplementations);

#define OPENSSL_TYPED_TEST(TestName, TestCase, CodeBlock) TYPED_TEST(TestName, TestCase) CodeBlock

OPENSSL_TYPED_TEST(OpenSSLTraceTest, TestOne, {
  printf("TestOne test");
})

```

Reply:
@oazizi000 please see how I implemented this in https://github.com/pixie-io/pixie/pull/419/commits/6f0fbf0f127fcd0fb854a015bd5f5cb669e8cd4d